Align bookmarks search design to address bar design

karlenDimla opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. 
The items in Bold are required
If your PR involves UI changes:
    1. Upload screenshots or screencasts that illustrate the changes before / after
    2. Add them under the UI changes section (feel free to add more columns if needed)
    3. Make sure these changes are tested in API 21 and API 26
If your PR does not involve UI changes, you can remove the **UI changes** section
-->

Task/Issue URL: https://app.asana.com/0/414730916066338/1200979549371790/f

### Description
This PR includes the following:
- Create a basic standalone SearchBar component that is meant to be shared between pages (not yet included in this PR)
- Integrate the SearchBar with the Bookmarks Activity replacing the Android SearchView widget

### Steps to test this PR

Scenario 1: User has no bookmarks
- Open bookmarks page
- Search icon should not be visible

Scenario 2: User has bookmarks
- Open bookmarks page
- Click on the Search icon
- The search bar should appear and it should look like the address bar in the browse page
- Typing on the search bar should filter results as the user types
- Pressing back clears and hides the search bar
- Pressing clear clears the search bar

### UI changes
| Before  | After |
| ------ | ----- |
|![before_bookmark_search](https://user-images.githubusercontent.com/2943941/152502898-5f85a074-acfd-49b6-863d-d8b0ae09f225.png)|![bookmark_search_clear](https://user-images.githubusercontent.com/2943941/152502964-654b16d2-4293-47a9-924e-c25ccf1cecb2.png)|
|![before_bookmark_search_result](https://user-images.githubusercontent.com/2943941/152503047-e390046e-e364-455f-b784-24317e535a07.png)|![bookmark_search_result](https://user-images.githubusercontent.com/2943941/152503157-a4bfe90d-8d23-49f2-a049-8d0e621290c4.png)|
|![before_bookmark_empty](https://user-images.githubusercontent.com/2943941/152503202-dfaf3b9f-f710-45dc-8a99-9532f36cca3a.png)|![bookmark_search_empty](https://user-images.githubusercontent.com/2943941/152503226-a92331e9-2a04-4cc6-94ff-7d6b47fceef2.png)|

joshliebe reviewed (commented):
Looks good so far 🙂

I gave it a try and the filtering isn't working like it did before though, it should search all bookmarks, even the ones in folders (So when the search is empty it should show all folders and bookmarks in the list).

Had a look at the `SearchBar`, I'm happy with the approach you're taking with the widget 👍

karlenDimla commented on app/src/main/java/com/duckduckgo/app/bookmarks/ui/SearchBar.kt:
```diff
@@ -0,0 +1,83 @@
+) : ConstraintLayout(context, attrs, defStyleAttr), SearchBar {
+    private val binding: ComponentSearchBarBinding by viewBinding()
+
+    init {
```
Ah no real benefit actually. I changed this part to be done through xml and styles.

karlenDimla commented:
I've fixed the incorrect behavior and moved the component to the common-ui module. I talked more to Sveta about this and we will get more feedback around this behavior before merging any changes. Just an FYI.

malmstein commented on app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt:
```diff
@@ -249,23 +255,43 @@ class BookmarksActivity : DuckDuckGoActivity() {
+    }
+
+    private fun showSearchBar() {
+        toolbar.visibility = GONE
```
```suggestion
        toolbar.gone()
```

malmstein commented on app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksActivity.kt:
```diff
@@ -249,23 +255,43 @@ class BookmarksActivity : DuckDuckGoActivity() {
+    }
+
+    private fun hideSearchBar() {
+        toolbar.visibility = VISIBLE
```
```suggestion
        toolbar.show()
```

malmstein commented on app/src/main/res/layout/activity_bookmarks.xml:
```diff
@@ -17,16 +17,44 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?toolbarBgColor"
+        android:theme="@style/ToolbarTheme">
```
```suggestion
        android:theme="@style/Widget.DuckDuckGo.ToolbarTheme">
```

malmstein commented on app/src/main/res/layout/activity_bookmarks.xml:
```diff
@@ -17,16 +17,44 @@
+        android:id="@+id/appBarLayout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?toolbarBgColor"
```
with the theme suggestion you can get rid of this

malmstein commented on app/src/main/res/layout/activity_bookmarks.xml:
```diff
@@ -17,16 +17,44 @@
+                android:layout_width="match_parent"
+                android:layout_height="?attr/actionBarSize"
+                android:background="?attr/toolbarBgColor"
+                android:theme="@style/ToolbarTheme"
```
```suggestion
                android:theme="@style/Widget.DuckDuckGo.ToolbarTheme"
```

malmstein commented on app/src/main/res/layout/activity_bookmarks.xml:
```diff
@@ -17,16 +17,44 @@
+                android:layout_height="?attr/actionBarSize"
+                android:background="?attr/toolbarBgColor"
+                android:theme="@style/ToolbarTheme"
+                app:popupTheme="@style/PopUpOverflowMenu" />
```
```suggestion
                app:popupTheme="@style/Widget.DuckDuckGo.PopUpOverflowMenu" />
```

malmstein commented on common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/SearchBar.kt:
```diff
@@ -82,6 +82,7 @@ class SearchBarView @JvmOverloads constructor(
             }
             SearchBar.Event.ShowSearchBar -> {
                 clearText()
+                binding.omnibarTextInput.showKeyboard()
```
this already calls `requestFocus()`, so you can get rid of the next line

malmstein reviewed (approved):
Couple of nits and suggestions, but this looks really good. Great work @karlenDimla!
